src/main.rs

```diff
@@ -1,3 +1,40 @@
+use std::collections::HashMap;
+use std::iter::FromIterator;
+
 fn main() {
-    println!("Hello, world!");
+    let mut bills = HashMap::new();
+    bills.insert("Niklas", 80f64);
+    bills.insert("Hampus", 200f64);
+    bills.insert("Magnus", 400f64);
+    bills.insert("Martin", 0f64);
+
+    let total_sum: f64 = bills.values().sum();
+    let total_sum = total_sum / bills.len() as f64;
+    bills.values_mut().for_each(|v| *v -= total_sum);
+
+    let (owed, owing): (HashMap<&str, f64>, HashMap<&str, f64>) =
+        bills.iter().partition(|(_, &v)| v >= 0.0);
+
+    let mut owed = Vec::from_iter(owed);
+    owed.sort_by(|(_, v1), (_, v2)| v2.partial_cmp(v1).expect("Can't sort"));
```

Comment:
This will just break? We'll have a `panic` which is hmm... Not good.
At least a better error message would be better. :+1: